Incomplete uninstallation via Python Install Manager: leftover shortcuts, registry entries, and internal AttributeError

Installed from the Windows Store
Version: 26
Windows 11

**Describe the bug**

**Python Install Manager** fails to fully uninstall Python versions. The uninstall process removes the main directory but leaves behind shortcuts, registry entries (the program remains listed in Windows "Uninstall or change a program"), and throws an internal `AttributeError` at the end of the process.

**Actual Behavior:**
```
C:\Windows\System32>py uninstall 3.12
Uninstall Python 3.12.10? [Y/n] y
Removed Python 3.12.10
[ERROR] INTERNAL ERROR: AttributeError: 'UninstallCommand' object has no attribute 'enable_entrypoints'
```
**After the process completes:**
C:\Users\<username>\AppData\Local\Python — Python files are removed ✅
Desktop/Start Menu shortcuts — still present ❌
Windows "Uninstall or change a program" entry — still present ❌

**Expected Behavior:**
A complete uninstall should remove all associated files, shortcuts, and registry entries without any internal errors.

**Additional Context:**
The same incomplete uninstallation occurs when using the standard Windows uninstaller via "Uninstall or change a program" - not just via `py uninstall`. This suggests the issue is systemic to how the Install Manager registers and cleans up components.

***
One possible workaround is to run `py uninstall --purge`. This isn’t really a proper solution (since this command removes all installed Python runtimes), but after running it all leftover traces were removed from the computer (shortcuts and entries in Uninstall or change a program). After that, I had to reinstall the Python runtimes I needed.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -140,6 +140,11 @@ jobs:
       env:
         PYMANAGER_DEBUG: true
 
+    - name: 'Install additional runtimes'
+      run: pymanager install 3.5 3.6 3.7 3.8 3.9
+      env:
+        PYMANAGER_DEBUG: true
+
     - name: 'List installed runtimes'
       run: pymanager list
       env:

--- a/ci/release.yml
+++ b/ci/release.yml
@@ -252,6 +252,13 @@ stages:
           PYMANAGER_DEBUG: true
 
       - powershell: |
+          pymanager install 3.5 3.6 3.7 3.8 3.9
+        displayName: 'Install additional runtimes'
+        timeoutInMinutes: 10
+        env:
+          PYMANAGER_DEBUG: true
+
+      - powershell: |
           pymanager list
         displayName: 'List installed runtimes'
         timeoutInMinutes: 1

--- a/src/manage/commands.py
+++ b/src/manage/commands.py
@@ -902,12 +902,25 @@ Removes one or more runtimes from your machine.
     purge = False
     by_id = False
 
-    # Not settable, but are checked by update_all_shortcuts() so we need them.
+    # Not directly settable, but will be copied from install configuration
     enable_shortcut_kinds = None
     disable_shortcut_kinds = None
+    enable_entrypoints = True
+    hard_link_entrypoints = True
 
     def execute(self):
         from .uninstall_command import execute
+        # Copy settings from install section of config
+        for k in [
+            "enable_shortcut_kinds",
+            "disable_shortcut_kinds",
+            "enable_entrypoints",
+            "hard_link_entrypoints",
+        ]:
+            v = self.config.get("install", {}).get(k, ...)
+            if v is not ...:
+                setattr(self, k, v)
+                LOGGER.debug("Setting config %s to %s from install.%s", k, v, k)
         self.show_welcome()
         execute(self)
 

